src/formatDate.js

```diff
@@ -50,7 +50,39 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const formatLength = fromFormat.length;
+  const separatorFrom = fromFormat[formatLength - 1];
+  const splitedDate = date.split(separatorFrom);
+  const dataObj = {};
+
+  for (let i = 0; i < formatLength - 1; i++) {
+    dataObj[fromFormat[i]] = splitedDate[i];
+  }
+
+  const yearFormatCheck = dataObj.hasOwnProperty('YY');
+
+  switch (true) {
+    case yearFormatCheck:
+      dataObj.YYYY = +dataObj.YY < 30
+        ? '20' + dataObj.YY
+        : dataObj.YYYY = '19' + dataObj.YY;
+      break;
```

Comment:
Looks really short, but it affects the readability

Reply:
dataObj.YYYY = '19' + dataObj.YY; - I find mistake)) I don't need to write here in variable